File: 프로그래머스/lv2/12945. 피보나치 수/피보나치 수.swift
func solution(_ n:Int) -> Int {
    var arr: [Int] = []
    
    arr.append(0)
    arr.append(1)
    
    for i in 2...n + 2 {
        arr.append((arr[i - 1] + arr[i - 2]) % 1234567)
    }
    return arr[n]
}

File: 백준/Bronze/2747. 피보나치 수/피보나치 수.cc
#include <iostream>
using namespace std;

int dp[46] = {0, 1, 1};

int solution(int n) {
    if(n == 0) return 0;
    else if (!dp[n]) 
        dp[n] = solution(n - 1) + solution(n - 2);
    return dp[n];
}

int main() {
  int n;
  cin >> n;
  
  cout << solution(n);
  
  return 0;
}